### PR TITLE
Fix fast leaves when texture_clean_transparent is enabled.

### PR DESCRIPTION
**Pull request by Warr1024**

### CODE REVIEW

File: src/client/tile.cpp [hunk 1]
```diff
@@ -1658,7 +1654,11 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			blit_with_interpolate_overlay(img, baseimg, v2s32(0,0), v2s32(0,0), dim, ratio);
 			img->drop();
 		}
-		else if (part_of_name.substr(0,19) == "[autoupscaleformesh") {
+		else if (part_of_name.substr(0,20) == "[applyfiltersformesh") {
```
Comment (Zeno-) [inline src/client/tile.cpp]: str_starts_with() (utils/string.h)

**Comment by est31** [comment]
Could you please post screenshots of how this improves?

**Comment by Warr1024** [comment]
Enable texture_clean_transparent and disable Fancy Leaves.

What it looks like in current master:
screenshot_20150424_144702

What it's supposed to look like:
screenshot_20150424_144011

File: src/client/tile.cpp [hunk 2]
```diff
@@ -1657,7 +1654,12 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			blit_with_interpolate_overlay(img, baseimg, v2s32(0,0), v2s32(0,0), dim, ratio);
 			img->drop();
 		}
-		else if (part_of_name.substr(0,19) == "[autoupscaleformesh") {
+		else if (str_starts_with(part_of_name, "[applyfiltersformesh"))
```
Comment (est31) [inline src/client/tile.cpp]: bracket on same line please

**Comment by HybridDog** [comment]
@Warr1024 ln my opinion the first one looks better.

**Comment by est31** [comment]
:+1: if style is fixed as outlined above.

**Comment by C1ffisme** [comment]
Yeah, I like screenshot 1.

**Comment by Zeno-** [comment]
Merging without the formatting pointed at by @est31 (because the rest of the else-else if-else construct is using the same format as the patch)

1fb3d1156cfdb2e3b35b20f84099ec5bdde172ae

**Comment by est31** [comment]
The whole if-else if- construct already has been touched by the `str_starts_with(part_of_name,` changes.

**Pull request closed without merging**